### PR TITLE
change quniform/qloguniform style in GridSearch to conform with other tuners

### PR DESCRIPTION
**Pull request by suiguoxin**
issue #1335

### CODE REVIEW

File: src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py [hunk 1]
```diff
@@ -43,9 +43,8 @@ class GridSearchTuner(Tuner):
 
     Type 'choice' will select one of the options. Note that it can also be nested.
 
-    Type 'quniform' will receive three values [low, high, q], where [low, high] specifies a range and 'q' specifies the number of values that will be sampled evenly.
-    Note that q should be at least 2.
-    It will be sampled in a way that the first sampled value is 'low', and each of the following values is (high-low)/q larger that the value in front of it.
+    Type 'quniform' will receive three values [low, high, q], where [low, high] specifies a range and 'q' specifies the interval
+    It will be sampled in a way that the first sampled value is 'low', and each of the following values is 'interval' larger that the value in front of it.
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py]: larger than

File: src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py [hunk 2]
```diff
@@ -95,19 +94,16 @@ def json2parameter(self, ss_spec):
...
+        return [low + interval * i for i in range(count)]
 
     def parse_qtype(self, param_type, param_value):
         '''parse type of quniform or qloguniform'''
-        if param_type == 'quniform':
-            return self._parse_quniform(param_value)
-        if param_type == 'qloguniform':
-            param_value[:2] = np.log(param_value[:2])
-            return list(np.exp(self._parse_quniform(param_value)))
+        if param_type in ['quniform', 'qloguniform']:
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py]: cannot deal with quniform and qloguniform in the same way

**Comment by QuanluZhang** [comment]
please update Grid Search in BuiltinTuner.md

File: src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py (hunk 2 above)
Comment (suiguoxin) [reply src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py]: what's the difference ?

File: src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py (hunk 1 above)
Comment (suiguoxin) [reply src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py]: fixed

**Comment by suiguoxin** [comment]
> please update Grid Search in BuiltinTuner.md

updated

**Comment by suiguoxin** [comment]
new modif:
- delete support for "qloguniform"
- add support for "randint"

File: src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py [hunk 3]
```diff
@@ -39,16 +39,14 @@ class GridSearchTuner
...
 
-    Type 'quniform' will receive three values [low, high, q], where [low, high] specifies a range and 'q' specifies the number of values that will be sampled evenly.
-    Note that q should be at least 2.
-    It will be sampled in a way that the first sampled value is 'low', and each of the following values is (high-low)/q larger that the value in front of it.
+    Type 'quniform' will receive three values [low, high, q], where [low, high] specifies a range and 'q' specifies the interval
+    It will be sampled in a way that the first sampled value is 'low', and each of the following values is 'interval' larger than the value in front of it.
 
-    Type 'qloguniform' behaves like 'quniform' except that it will first change the range to [log(low), log(high)]
-    and sample and then change the sampled value back.
+    Type 'randint' gives all possible intergers in range(low, high). Note that 'high' is not included.
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py]: range(low, high) => [low, high), more clear?

**Comment by QuanluZhang** [comment]
looks good

File: src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py (hunk 3 above)
Comment (suiguoxin) [reply src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py]: updated

**Pull request merged**